Week1/homework/world-queries.txt

```diff
@@ -37,4 +37,153 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
+
+select name from country where continent='North America';
+
+name
+------
+Aruba
+Anguilla
+Netherlands Antilles
+Antigua and Barbuda
+Bahamas
+Belize
+Bermuda
+Barbados
+Canada
+Costa Rica
+Cuba
+Cayman Islands
+Dominica
+Dominican Republic
+Guadeloupe
+Grenada
+Greenland
+Guatemala
+Honduras
+Haiti
+Jamaica
+Saint Kitts and Nevis
+Saint Lucia
+Mexico
+Montserrat
+Martinique
+Nicaragua
+Panama
+Puerto Rico
+El Salvador
+Saint Pierre and Miquelon
+Turks and Caicos Islands
+Trinidad and Tobago
+United States
+Saint Vincent and the Grenadines
+Virgin Islands, British
+Virgin Islands, U.S.
+
+4. What's the top 10 countries by SurfaceArea ?
+
+SELECT Name, SurfaceArea FROM country ORDER BY SurfaceArea DESC LIMIT 10;
+
+Name	             SurfaceArea
+---------------------------------
+Russian Federation	   17075400
+Antarctica             13120000
+Canada	               9970610
+China	               9572900
+United States	       9363520
+Brazil	               8547403
+Australia	           7741220
+India	               3287263
+Argentina	           2780400
+Kazakstan	           2724900
+
+5. What's the top 10 most populated cities?
+
+select name,countrycode,population from city order by population desc limit 10;
+
+name	    countrycode	   population
+-------------------------------------
+Mumbai(Bombay)	IND	        10500000
+Seoul	        KOR     	9981619
+SÃ£o Paulo	    BRA	        9968485
+Shanghai	    CHN	        9696300
+Jakarta	        IDN	        9604900
+Karachi	        PAK	        9269265
+Istanbul	    TUR	        8787958
+Ciudad de MÃ©xicoMEX	    8591309
+Moscow	        RUS	        8389200
+New York	    USA	        8008278
+
+6.Select the top 3 countries by population that their names start with the letter ‘P’
+
+select name,population from country where name like 'p%' order by population desc limit 3;
+
+name	     population
+------------------------
+Pakistan	 156483000
+Philippines	 75967000
+Poland	     38653600
+
+7. What is the population in Asia?
+
+select sum(population) as "Asia's Popluation" from country where continent='Asia';
+
+Asia's Popluation
+-----------------
+3705025700
+
+8. Which languages are spoken in the Region of ‘South America’
+
+SELECT distinct language as "Languages Spoken in South America" FROM countrylanguage INNER JOIN country ON country.code=countrylanguage.countrycode WHERE country.continent='South America';
+
+Languages Spoken in South America
+-----------------------------------
+Indian Languages
+Italian
+Spanish
+AimarÃ¡
+GuaranÃ­
+KetÂšua
+German
+Japanese
+Portuguese
+Araucan
+Rapa nui
+Arawakan
+Caribbean
+Chibcha
+Creole English
+English
+Creole French
+Hindi
+Sranantonga
+Goajiro
+Warrau
+
+9. What are the languages spoken on all cities named ‘Barcelona’
+
+SELECT city.Name, countrylanguage.language FROM countrylanguage JOIN city ON city.Countrycode = countrylanguage.Countrycode JOIN country ON city.Countrycode = country.Code WHERE city.Name = 'Barcelona';
```

Comment:
Nice job 👏  (query did not specify what to return) but it would be nice to get the country name as well to satisfy my curiosity... is there more than one Barcelona ? 😝  

Also since you clearly know your sql feel free to go crazy with aliases to avoid writing the whole table name each time .. like..

```
SELECT
  ci.name,
  co.name,
  l.language
FROM country co
  JOIN city ci on ci.countrycode = co.Code
  JOIN countrylanguage l ON l.CountryCode = co.Code
WHERE ci.Name = 'Barcelona';
```

Reply:
done !